### PR TITLE
4.x: Cleaner typehinting for Folder.

### PR DESCRIPTION
**Pull request by dereuromark**
Further cleanup

### CODE REVIEW

File: tests/TestCase/Filesystem/FolderTest.php [hunk 1]
```diff
@@ -277,11 +277,11 @@ public function testOperations()
         $this->assertTrue($result);
 
         $copy = TMP . 'tests' . DS . 'test_folder_copy';
-        $result = $Folder->copy($copy);
+        $result = $Folder->copy(['to' => $copy]);
```
Comment (ADmad) [inline tests/TestCase/Filesystem/FolderTest.php]: This is not a convenient usage. In most cases one would just specify destination path without any options. So changing the method signature to `copy(string $dest, array $options)` would be better.

**Comment by dereuromark** [comment]
As own param this is indeed much cleaner. This argument was required anyway, but inside the options array rather hidden, now it is explicit.

**Comment by ADmad** [comment]
`strict_types` declaration missing at top of file.

**Comment by dereuromark** [comment]
Shouldn't we do that with a fixer instead of manually in files?

**Comment by ADmad** [comment]
That's an option too.

**Pull request merged**